mlir/lib/Dialect/MQTOpt/Transforms/ElidePermutationsPattern.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+ * Copyright (c) 2025 Munich Quantum Software Company GmbH
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Licensed under the MIT License
+ */
+
+#include "mlir/Dialect/MQTOpt/IR/MQTOptDialect.h"
+#include "mlir/Dialect/MQTOpt/Transforms/Passes.h"
+
+#include <llvm/ADT/STLExtras.h>
+#include <mlir/IR/MLIRContext.h>
+#include <mlir/IR/PatternMatch.h>
+#include <mlir/IR/Value.h>
+#include <mlir/IR/ValueRange.h>
+#include <mlir/Support/LLVM.h>
+#include <mlir/Support/LogicalResult.h>
+#include <vector>
+
+namespace mqt::ir::opt {
+
+/**
+ * @brief This pattern attempts to remove SWAP gates by re-ordering qubits.
+ */
+struct ElidePermutationsPattern final : mlir::OpRewritePattern<SWAPOp> {
+
+  explicit ElidePermutationsPattern(mlir::MLIRContext* context)
+      : OpRewritePattern(context) {}
+
+  mlir::LogicalResult
+  matchAndRewrite(SWAPOp op, mlir::PatternRewriter& rewriter) const override {
+    const bool isControlledSwap = !op.getAllCtrlInQubits().empty();
+    if (isControlledSwap) {
```

Comment:
Best case, this simply becomes 
```suggestion
    if (op.isControlled()) {
```
after addressing the comment above.